test/fixtures/array-func/avoid-reverse.fail.ts

```diff
@@ -1,5 +1,6 @@
 /* eslint-disable @typescript-eslint/explicit-function-return-type */
 /* eslint-disable id-length */
+/* eslint-disable unicorn/no-array-reverse */
```

Comment:
there's a proposal for `prefer-array-last-methods` in `eslint-plugin-unicorn` that would support suggesting to change `.toReversed().reduce()` to `.reduceRight()`, but since it's not there yet it looks to me like we should keep both the `array-func/avoid-reverse` rule (that suggests .reduceRight for example) and the `unicorn/no-array-reverse` - to suggest `.toReversed()` instead of `.reverse()` in general as it's a non-mutating one